invertedai/utils.py

```diff
@@ -563,6 +575,211 @@ def inside_fov(center: Point, initialize_fov: float, point: Point) -> bool:
+        agent_attributes_con = agent_attributes_con if len(agent_attributes_con) > 0 else None
+
+        agent_density_area_center = agent_density_list[i]
+
+        # Check if this initialization call can be skipped
+        if len(agent_attributes_specified_area) < 1:
+            # Only if there are no conditional agents in the area can it be skipped.
+            if agent_density_area_center < 1:
+                #Skip if no agents are requested (e.g. in areas with no drivable surfaces)
+                continue
+            elif len(con_agent_state) >= agent_density_area_center:
+                #Skip if the calculated number of agents has been satisfied or surpassed
+                continue
+
+        for _ in range(1):
+            try:
+                # Initialize simulation with an API call
+                response = iai.initialize(
+                    location=location,
+                    states_history=states_history,
+                    agent_attributes=agent_attributes_con,
+                    agent_count=agent_density_area_center,
+                    get_infractions=False,
+                    traffic_light_state_history=traffic_lights_states,
+                    location_of_interest=(area_center.x, area_center.y),
+                    random_seed=random_seed,
+                    get_birdview=get_birdview,
+                )
+
+                if traffic_lights_states is None:
+                    # If no traffic light states are given, take the first non-None traffic light states output as the consistent traffic light states across all areas
+                    traffic_lights_states = [response.traffic_lights_states]
+
+                break
+            except BaseException as e:
+                print(e)
+        else:
+            continue
+
+        # Get the recurrent state for any conditional agents in this region
+        for s, rs in zip(response.agent_states[:len(states_history_specified_area)],response.recurrent_states[:len(states_history_specified_area)]):
+            conditional_agent_recurrent_state_dict[(s.center.x,s.center.y)] = rs
+
+        # Remove all conditional agents before filtering at the edges
+        non_conditional_agent_states = response.agent_states[len(states_history_specified_area):]
+        non_conditional_agent_attributes = response.agent_attributes[len(agent_attributes_specified_area):]
+        non_conditional_recurrent_states = response.recurrent_states[len(states_history_specified_area):]
+
+        # Filter out agents that are not inside the ROI to avoid collision with other agents not passed as conditional
+        # SLACK is for removing the agents that are very close to the boundary and
+        # they may collide agents not filtered as conditional
+        valid_agents = list(filter(
+            lambda x: inside_fov(center=area_center, initialize_fov=initialize_fov - SLACK, point=x[0].center),
```

Comment:
```suggestion
            lambda x: inside_fov(center=area_center, initialize_fov=min(2*stride, initialize_fov - SLACK), point=x[0].center),
```
Based on the assumption that INITIALIZE works better near the center, where it has more context (maybe @bzwartsenberg can speak to whether that's true) it would make sense to crop tighter and let each region be primarily initialized by the call with the closest center.

Reply:
Why would we allow the region in which the function considers conditional agents to be potentially limited by the user setting the stride parameter? For example, if the user sets the parameter to 25, initialize() will potentially produce collisions within the outer limit of the region.

Reply:
Can you elaborate what you mean by "near the center", near the center of what specifically? Near the center of an arbitrary 100x100m space or near the center of an entire map?

Reply:
I think I understand what you mean. A vehicle should only be considered valid if the closest region center was the region of interest from which it was sampled, correct? I think that could represent some significant changes to how area_initialize() works beyond tuning the SLACK value (e.g. how do we find/deal with regions on the edge of drivable areas? If we prune too many agents, how will that impact remaining agents behaviour?). My opinion is that it should be a PR for another time since my best guess is that it will not impact the user interface and can be a performance improvement we roll out quietly at a later date.